pkg/devfile/validate/validate.go

```diff
@@ -14,16 +13,13 @@ import (
 func ValidateDevfileData(data interface{}) error {
 	var components []common.DevfileComponent
 
-	typeData := reflect.TypeOf(data)
-
-	if typeData == reflect.TypeOf(&v100.Devfile100{}) {
-		d := data.(*v100.Devfile100)
+	switch d := data.(type) {
+	case *v100.Devfile100:
```

Comment:
this can be adjusted in a single case clause
```suggestion
	case *v100.Devfile100, *v200.Devfile200:
	         components := d.GetComponents( )
	
```

Reply:
I tried that, this won't work because go != java  and `Devfile100` & `Devfile200` don't share a common interface (which might actually be something worth exploring provided we actually still care about devfile 1.0). I would have done this change if it was possible to do so… 😄

Reply:
oh I see.

my bad :sweat_smile: